src/redux/books/books.js

```diff
@@ -0,0 +1,34 @@
+// Actions
+const ADD = 'bookstore/books/add';
+const DEL = 'bookstore/books/del';
+
+// Empty array of books
+const initialState = [];
+
+// Reducer
+export default function booksReducer(state = initialState, action = {}) {
+  switch (action.type) {
+    case ADD:
+      return [...state, {
+        ...action.payload,
+      }];
+    case DEL:
+      return state.filter((book) => book.id !== action.id);
+    default: return state;
+  }
+}
+
+// Action Creators
+export function add(book) {
+  return {
+    type: ADD,
+    payload: { book },
+  };
```

Comment:
- [ ] Kindly make sure your newly added book will be iterable very easily it is not necessary to wrap it into an object because it's self will be an object.

  _**Suggestions:**_ ` return { type: ADD, payload: book }`